apps/api/src/app/widgets/usecases/mark-all-messages-as/mark-all-messages-as.usecase.ts

```diff
@@ -0,0 +1,66 @@
+import { Inject, Injectable, NotFoundException } from '@nestjs/common';
+import { MessageRepository, SubscriberRepository } from '@novu/dal';
+import { AnalyticsService } from '@novu/application-generic';
+
+import { CacheKeyPrefixEnum } from '../../../shared/services/cache';
+import { QueueService } from '../../../shared/services/queue';
+import { ANALYTICS_SERVICE } from '../../../shared/shared.module';
+import { InvalidateCache } from '../../../shared/interceptors';
+import { MarkAllMessagesAsCommand } from './mark-all-messages-as.command';
+
+@Injectable()
+export class MarkAllMessagesAs {
+  constructor(
+    private messageRepository: MessageRepository,
+    private queueService: QueueService,
+    private subscriberRepository: SubscriberRepository,
+    @Inject(ANALYTICS_SERVICE) private analyticsService: AnalyticsService
+  ) {}
+
+  @InvalidateCache([CacheKeyPrefixEnum.MESSAGE_COUNT, CacheKeyPrefixEnum.FEED])
+  async execute(command: MarkAllMessagesAsCommand): Promise<number> {
+    const subscriber = await this.subscriberRepository.findBySubscriberId(command.environmentId, command.subscriberId);
+    if (!subscriber) {
+      throw new NotFoundException(
+        `Subscriber ${command.subscriberId} does not exist in environment ${command.environmentId}, ` +
+          `please provide a valid subscriber identifier`
+      );
+    }
+
+    const response = await this.messageRepository.markAllMessagesAs(
+      subscriber._id,
+      command.environmentId,
+      command.markAs,
+      command.feedIds
+    );
+
+    this.queueService.wsSocketQueue.add({
+      event: 'unseen_count_changed',
+      userId: subscriber._id,
+      payload: {
+        unseenCount: 0,
+      },
+    });
+
+    await this.queueService.wsSocketQueue.add({
+      event: 'unread_count_changed',
+      userId: subscriber._id,
+      payload: {
+        unreadCount: 0,
+      },
+    });
+
```

Comment:
Shouldn't we conditionaly dire those depending on the `command.markAs` ?

Reply:
ah yes, added them